### PR TITLE
Domain owners signal

### PR DESCRIPTION
**Pull request by ghost**
Send domain data during save through `Hermes`.

### CODE REVIEW

File: src/ralph/domains/publishers.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    @pyhermes.publisher(
+        topic=settings.DNSAAS_DOMAIN_SERVICE_UPDATE_TOPIC,
+        auto_publish_result=True
+    )
+    def publish_domain_data_to_dnsaas(obj):
+        return _publish_domain_to_dnsaaas(obj)
+
+    @receiver(post_save, sender=Domain)
+    def post_save_domain(sender, instance, **kwargs):
+        publish_domain_data_to_dnsaas(instance)
```
Comment (szok) [inline src/ralph/domains/publishers.py]: Please change function name to  more general

File: src/ralph/settings/base.py [hunk 2]
```diff
@@ -432,6 +432,14 @@ def str_to_bool(s: str) -> bool:
     )
 DNSAAS_AUTO_UPDATE_HOST_DNS = bool_from_env('DNSAAS_AUTO_UPDATE_HOST_DNS')
 
+DOMAIN_DATA_UPDATE_TOPIC = bool_from_env(
```
Comment (mkurek) [inline src/ralph/settings/base.py]: if it's bool, how this could be set to hermes topic name?

File: src/ralph/domains/publishers.py [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+        })
+
+    if not owners:
+        logger.debug('no owners for domain: {}'.format(domain.id))
+        return {}
+
+    domain_data = {
+        'domain_name': domain.name,
+        'owners': owners,
+    }
```
Comment (mkurek) [inline src/ralph/domains/publishers.py:32]: what about service (uid)?

File: src/ralph/domains/publishers.py [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+        })
+    if domain.technical_owner:
+        owners.append({
+            'username': domain.technical_owner.username,
+            'ownership_type': settings.DOMAIN_OWNER_TYPE['TO'],
+        })
+
+    if not owners:
+        logger.debug('no owners for domain: {}'.format(domain.id))
+        return {}
```
Comment (mkurek) [inline src/ralph/domains/publishers.py]: that's fine in my opinion - i'd send to hermes info about empty owners as well

**Review by mkurek — CHANGES_REQUESTED** [review]

File: src/ralph/domains/publishers.py [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+        })
+    if domain.technical_owner:
+        owners.append({
+            'username': domain.technical_owner.username,
+            'ownership_type': settings.DOMAIN_OWNER_TYPE['TO'],
+        })
+
+    domain_data = {
+        'domain_name': domain.name,
+        'service_uid': domain.service.uid,
```
Comment (mkurek) [inline src/ralph/domains/publishers.py]: what if domain has no service?

**Review by mkurek — CHANGES_REQUESTED** [review]

File: src/ralph/domains/publishers.py [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+        })
+    if domain.technical_owner:
+        owners.append({
+            'username': domain.technical_owner.username,
+            'ownership_type': settings.DOMAIN_OWNER_TYPE['TO'],
+        })
+
+    domain_data = {
+        'domain_name': domain.name,
+        'service_uid': domain.service.uid if domain.service else '',
```
Comment (mkurek) [inline src/ralph/domains/publishers.py:30]: any tests for case with domain without service?

Comment (ghost) [reply src/ralph/domains/publishers.py:30]: https://github.com/allegro/ralph/pull/2854/files#diff-adc5af73512606f536b74d9c4b032988R101

**Review by mkurek — APPROVED** [review]

**Pull request merged**